messenger/make.py

```diff
@@ -1,71 +1,104 @@
 #!/usr/bin/python
 from __future__ import print_function
 import os
-from get_messenger_dir import get_messenger_dir
-import subprocess
+import platform
+import sys
+import shlex
 import shutil
+import subprocess
+
+
+def get_messenger_dir():
+    # Check the system platform first
+    splatform = sys.platform
+
+    if splatform.startswith('linux'):
+        messenger_dir = 'mexa64'
+    elif splatform.startswith('darwin'):
+        messenger_dir = 'mexmaci64'
+    elif splatform.startswith('win32'):
+        if splatform == "win32":
+            # We have a win64 messenger, so we need to figure out if this is 32
+            # or 64 bit Windows:
+            if not platform.machine().endswith('64'):
+                raise ValueError("pymatbridge does not work on win32")
+
+        # We further need to differniate 32 from 64 bit:
+        maxint = sys.maxsize
+        if maxint == 9223372036854775807:
+            messenger_dir = 'mexw64'
+        elif maxint == 2147483647:
+            messenger_dir = 'mexw32'
+    return messenger_dir
+
+
+def get_config():
+    messenger_dir = get_messenger_dir()
+    with open(os.path.join(messenger_dir, 'local.cfg')) as fid:
+        lines = fid.readlines()
+
+    cfg = {}
+    for line in lines:
+        if '=' not in line:
+            continue
+        name, path = line.split('=')
+        cfg[name.lower()] = path.strip() or '.'
+    return cfg
+
+
+def do_build(make_cmd, messenger_exe):
+    print('Building %s...' % messenger_exe)
+    print(make_cmd)
+    messenger_dir = get_messenger_dir()
+    subprocess.check_output(shlex.split(make_cmd))
+
+    messenger_loc = os.path.join(messenger_dir, messenger_exe)
+
+    shutil.move(messenger_exe, messenger_loc)
+
+    try:
+        os.remove('messenger.o')
+    except OSError:
+        os.remove('src/messenger.o')
```

Comment:
I still get an error here when building for matlab. I ran the mex command manually and it looks like I don't get a `.o` file anywhere, actually.